com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java

```diff
@@ -93,30 +92,16 @@ static CloudControllerClient getControllerClient(DelegateExecution context, Clou
     }
 
     public static String determineCurrentUser(VariableScope scope, StepLogger stepLogger) {
-        String userId = Authentication.getAuthenticatedUserId();

```

Comment:
Why the usage of the Authentication.getAuthenticatedUserId is no more needed?
The Authentication.getAuthenticatedUserId is set by each command that is being executed over the flowable engine - any method in the FlowableFacade is using it.
Something more, if there is no more need of this, consider changing the FlowableFacade not to set it.

Reply:
Even though the methods in FlowableFacade set the authenticated user id, the object returned from Authentication.getAuthenticatedUserId is not null only on the first invocation (on process start). At least that's what the results of my debugging proved. That is why I removed its usage in determineCurrentUser. The reason that I left the variable setting in FlowableFacade is so that, as you said, to let flowable use it internally.